### PR TITLE
Convert more logs to json format and added manditory fields to log

### PR DESCRIPTION
**Pull request by sathishavm**
Convert logs to JSON format and added some mandatory fields to log.

/cc @zendesk/samson

### References
 - Jira link: https://zendesk.atlassian.net/browse/BRE-979

### Risks
- Level: Med

### CODE REVIEW

File: lib/samson/syslog_formatters.rb [hunk 1]
```diff
@@ -4,11 +4,20 @@
 module Samson
   class SyslogFormatter < Syslog::Logger::Formatter
     def call(severity, timestamp, _progname, message)
+      message_h =
+        begin
+          message = message.to_json unless message.is_a?(String)
+          # Fix for Integer and Array as a return object
```
[the remark this answers is not in the document]
Comment (sathishavm) [reply lib/samson/syslog_formatters.rb]: Right, We allow strings / hashes, here we just parse the hash and merge into the default log hash.

File: lib/samson/syslog_formatters.rb [hunk 2]
```diff
@@ -4,11 +4,21 @@
 module Samson
   class SyslogFormatter < Syslog::Logger::Formatter
     def call(severity, timestamp, _progname, message)
+      message_h =
+        begin
+          message = message.to_json unless message.is_a?(String)
```
Comment (grosser) [inline lib/samson/syslog_formatters.rb]: so message comes in as a string or hash ? ... blow up when it's anything else ?
.... does it ever come in as string with json ? / can we avoid that ?

Comment (sathishavm) [reply lib/samson/syslog_formatters.rb]: Not all times, we may also get a different object as input. 
I wish not to blow up because this formatter is now used by external gems, we just try to prettify the messages as much as possible, if not then we just squish them and embed in the message attributes of hash.

Comment (grosser) [reply lib/samson/syslog_formatters.rb]: converting things to json and then parsing them right after is inefficient, how can we avoid that ?
who is passing json-ified objects into the logger / can we opt out of supporting that ?

Comment (grosser) [reply lib/samson/syslog_formatters.rb]: why do we have to squish things ? .... that has the potential to modify important message details, so I'd rather avoid that if possible

Comment (sathishavm) [reply lib/samson/syslog_formatters.rb]: ok, made alterations.

File: lib/samson/logging.rb [hunk 3]
```diff
@@ -19,14 +19,14 @@
     {
       params: params,
       user_id: request.env['warden']&.user&.id,
-      ip: request.remote_ip
+      client_ip: request.remote_ip
```
Comment (grosser) [inline lib/samson/logging.rb]: why did this change, just readability or something else ?

File: lib/samson/syslog_formatters.rb [hunk 4]
```diff
@@ -4,11 +4,25 @@
 module Samson
   class SyslogFormatter < Syslog::Logger::Formatter
     def call(severity, timestamp, _progname, message)
+      message_h =
+        begin
```
Comment (grosser) [inline lib/samson/syslog_formatters.rb]: raising on every non-hash/json passed in is wasteful ... also code is kinda convoluted
better to dumb it down to this which not as dry but easier to read I hope:

```Ruby
if message.is_a?(Hash)
  message
elsif message.is_a?(String) && message.start_with?('{')
  begin
    JSON.parse(message)
  rescue JSON::ParserError
    {message: message}
  end
else
  {message: message}
end

```

File: lib/samson/logging.rb (hunk 3 above)
Comment (dragonfax) [reply lib/samson/logging.rb]: I would guess readability. I mean, I wouldn't know looking at the logs if it was the server's IP (so you can track down which server is bad) or the clients IP (who causing the problem).

File: lib/samson/syslog_formatters.rb (hunk 4 above)
Comment (grosser) [reply lib/samson/syslog_formatters.rb]: old: raise on any non-json string + on any non-string-and-non-hash
new: raise when a string looks like json but is not parseable

Comment (sathishavm) [reply lib/samson/syslog_formatters.rb]: fine, altered!!
